cmd/gh/main_test.go

```diff
@@ -76,3 +88,128 @@ check your internet connection or https://githubstatus.com
 		})
 	}
 }
+
+func Test_IntegrationRepoFork(t *testing.T) {
+	// Set up temp dir
+	tempDir := t.TempDir()
+	oldWd, _ := os.Getwd()
+	assert.NoError(t, os.Chdir(tempDir))
+	t.Cleanup(func() { _ = os.Chdir(oldWd) })
+
+	// Set up server to handle network requests
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		path := r.URL.Path
+		gitPath := regexp.MustCompile(`\/someone\/repo\.git\/.*`)
+		forkPath := "/repos/owner/repo/forks"
+		switch {
+		case path == forkPath:
+			forkResult := fmt.Sprintf(`{
+				"node_id": "123",
+				"name": "repo",
+				"clone_url": "https://github.com/someone/repo.git",
+				"created_at": "%s",
+				"owner": {
+					"login": "someone"
+				}
+			}`, time.Now().Format(time.RFC3339))
+			_, _ = w.Write([]byte(forkResult))
+		case gitPath.MatchString(path):
+			w.WriteHeader(http.StatusOK)
+		default:
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	}))
+	t.Cleanup(s.Close)
+	os.Setenv("HTTP_PROXY", s.URL)
+	t.Cleanup(func() { os.Unsetenv("HTTP_PROXY") })
+
+	// Set up config directory and config files
+	os.Setenv("GH_CONFIG_DIR", tempDir)
+	t.Cleanup(func() { os.Unsetenv("GH_CONFIG_DIR") })
+	hosts := heredoc.Doc(`
+    github.localhost:
```

Comment:
Have to use `github.localhost` as it is the only host that we don't elevate to `https`. With `https` the test server has to deal with certificates and other weird stuff so I avoided that hassle.